chill-java/src/main/java/com/twitter/chill/java/BitSetSerializer.java

```diff
@@ -24,32 +25,80 @@
 import com.twitter.chill.IKryoRegistrar;
 import com.twitter.chill.SingleRegistrar;
 
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
 import java.util.BitSet;
 
-public class BitSetSerializer extends Serializer<BitSet> {
+public class BitSetSerializer extends Serializer<BitSet> implements Serializable {
 
     static public IKryoRegistrar registrar() {
       return new SingleRegistrar(BitSet.class, new BitSetSerializer());
     }
 
+    private final static Field wordsField;
+    private final static Constructor<BitSet> bitSetConstructor;
+    private final static Method recalculateWordsInUseMethod;
+
+    static {
+        try {
+            wordsField = BitSet.class.getDeclaredField("words");
+            wordsField.setAccessible(true);
+        } catch (NoSuchFieldException e) {
+            throw new KryoException("Error while getting field 'words' of bitSet", e);
+        }
+        try {
+            bitSetConstructor = BitSet.class.getDeclaredConstructor(long[].class);
+            bitSetConstructor.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            throw new KryoException("Unable to get BitSet(long[]) constructor", e);
+        }
+        try {
+            recalculateWordsInUseMethod = BitSet.class.getDeclaredMethod("recalculateWordsInUse");
+            recalculateWordsInUseMethod.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            throw new KryoException("Unable to get BitSet.recalculateWordsInUse() method", e);
+        }
+    }
+
     @Override
     public void write(Kryo kryo, Output output, BitSet bitSet) {
-        int len = bitSet.length();
-
-        output.writeInt(len, true);
+        long words[] = null;
+        // its sufficent to get only the 'words' field because
+        // we can recompute the wordsInUse after deserialization
+        try {
+            words = (long[]) wordsField.get(bitSet);
+        } catch (IllegalAccessException e) {
+            throw new KryoException("Error while accessing field 'words' of bitSet", e);
+        }
+        output.writeInt(words.length, true);
 
-        for(int i = 0; i < len; i++) {
-            output.writeBoolean(bitSet.get(i));
+        for(int i = 0; i < words.length; i++) {
+            output.writeLong(words[i]);
```

Comment:
No, they are not. The `words` contain the the bits from the bitset. Depending on how the user sets the bits, they might represent very negative numbers ;)